src/ert/config/observations.py

```diff
@@ -1,471 +1,241 @@
-import os
-from datetime import datetime, timedelta
-from typing import TYPE_CHECKING, Dict, Iterator, List, Optional, Tuple, Union
+from dataclasses import dataclass, field
+from datetime import datetime
+from typing import (
+    Dict,
+    Iterator,
+    List,
+    Optional,
+    Protocol,
+    Union,
+)
 
-import numpy as np
+import pandas as pd
 import xarray as xr
 
-from ert.validation import rangestring_to_list
+from ert.config.summary_observation import SummaryObservation
 
-from .enkf_observation_implementation_type import EnkfObservationImplementationType
-from .gen_data_config import GenDataConfig
-from .general_observation import GenObservation
 from .observation_vector import ObsVector
-from .parsing import ConfigWarning, HistorySource
-from .parsing.observations_parser import (
-    DateValues,
-    ErrorValues,
-    GenObsValues,
-    HistoryValues,
-    ObservationConfigError,
-    SummaryValues,
-)
-from .summary_observation import SummaryObservation
-
-if TYPE_CHECKING:
-    import numpy.typing as npt
-
-    from .ensemble_config import EnsembleConfig
-
-DEFAULT_TIME_DELTA = timedelta(seconds=30)
+from .response_properties import ResponseTypes
 
 
 def history_key(key: str) -> str:
     keyword, *rest = key.split(":")
     return ":".join([keyword + "H"] + rest)
 
 
+class _AccumulatedDataset(Protocol):
+    def to_xarray(self) -> xr.Dataset: ...
+    def __len__(self) -> int: ...
+
+
+@dataclass
+class _SummaryObsDataset:
```

Comment:
How does this compare with `GenObservation`, and how should their relationship be?